tools/config-generator/main.go

```diff
@@ -945,6 +946,7 @@ func main() {
 	flag.StringVar(&jobNameFilter, "job-filter", "", "Generate only this job, instead of all jobs")
 	flag.StringVar(&preCommand, "pre-command", "", "Executable for running instead of the real command of a job")
 	flag.BoolVar(&upgradeReleaseBranches, "upgrade-release-branches", false, "Update release branches jobs based on active branches")
+	flag.StringVar(&githubTokenPath, "github-token-path", "", "Token path for authenticating with github, used only when --upgrade-release-branches is on")
```

Comment:
I tend to prefer interfaces and arguments which do not allow inconsistencies. A user could request `upgrade-release-branches` and not provide the token (or viceversa). Does it make sense to have a single program argument which encapsulates both options? E.g: `upgrade-release-branches-with-github-token` string

Reply:
To be honest, it makes sense to me combining these two params, I'm just not very comfortable with such a long param name

Reply:
We are fully aligned on that.
I'm open to both options, but lean a little bit more towards the single-parameter option.

Feel free to choose the one you prefer.